src/otx/algo/classification/heads/custom_multilabel_non_linear_cls_head.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""This module contains the CustomMultiLabelNonLinearClsHead implementation for MMClassification."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import torch
+from mmcv.cnn import build_activation_layer
+from mmengine.model import constant_init, normal_init
+from mmpretrain.models.heads import MultiLabelClsHead
+from mmpretrain.registry import MODELS
+from torch import nn
+
+from .custom_multilabel_linear_cls_head import AnglularLinear
+
+if TYPE_CHECKING:
+    from mmpretrain.structures import DataSample
+
+
+@MODELS.register_module()
+class CustomMultiLabelNonLinearClsHead(MultiLabelClsHead):
```

Comment:
In OTX1.X I thought that these two heads (NonLinear and Linear) can be merged into one class and initialized by a config.  The main obstacle was that we have one template for both tasks. In OTX2.0 we have a freedom to divide these two tasks into two templates, so I think now we can merge this into one class MultiLabelClsHead. 

It seems like the _init_layers() the main difference between two classification heads
![image](https://github.com/openvinotoolkit/training_extensions/assets/68103869/835deb8f-904a-4f8b-b393-7d09bb39256f)

Could you consider that?

Reply:
I could consider, however, is there no problem of the backward compatibility ? We've been using the `NonLinearHead` to the `MobileNet-v3-large-1.0` template for a long time. So, I have a concern about the backward compatibility

Reply:
Do we have there a problem with loading pre-trained weights if we change Class name? If it is the same -> then I think we can consider such change.

Reply:
Could I consider what you suggested at the later phase? I think integrating H-label classification is more important item.